go/tosca/types.go

```diff
@@ -79,6 +79,10 @@ func Sub(a, b Value) Value {
 	return ValueFromUint256(new(uint256.Int).Sub(a.ToUint256(), b.ToUint256()))
 }
 
+func Mul(a, b Value) Value {
```

Comment:
Something to be considered, but no need to address this in this PR: technically, Value is a number with a unit. The unit is "wei", and when multiplying two values with a unit, you would get the product of the units, in this case "wei^2", which is obviously a strange construct.

In reality, we probably never have any use case where we multiply two values with each other. What we actually do is scale a value by a unit-less scalar (e.g., GasPrice * Gas is a Value * scalar).

You may have noticed this problem because you have been forced to convert one of the parameters in the gas price computation to a value. This is a consequence of this. A way out, that at the first sight may look odd though, would be to define
```
func Mul(a Value, s uint64) Value { ... }
```
or
```
func (v Value) Scale(s uint64) Value { ... }
```
By defining it like this, it makes it easier to do the right operations (scaling a value) and harder to do something likely to be wrong (like multiplying values).

Reply:
Thank you for the remark. I think this is a good change because, as you mentioned, it makes it harder to do wrong things in the future. I changed it to the `Scale` function.